lib/omniauth/strategies/openid_connect.rb

```diff
@@ -255,8 +277,61 @@ def access_token
         @access_token
       end
 
+      # Unlike ::OpenIDConnect::ResponseObject::IdToken.decode, this
+      # method splits the decoding and verification of JWT into two
+      # steps. First, we decode the JWT without verifying it to
+      # determine the algorithm used to sign. Then, we verify it using
+      # the appropriate public key (e.g. if algorithm is RS256) or
+      # shared secret (e.g. if algorithm is HS256).  This works around a
+      # limitation in the openid_connect gem:
+      # https://github.com/nov/openid_connect/issues/61
```

Comment:
This issue appears to have been closed via https://github.com/nov/openid_connect/commit/17cc91eead594e6ce67f2db522d911b8db652743, but it still requires that the client know ahead of time which encryption mode is being used.